syntax = "proto3";

package contextworker;

import "context_unit.proto";

// Background task executor and orchestrator
service WorkerService {
    // Start a durable workflow process
    rpc StartWorkflow(contextcore.ContextUnit) returns (contextcore.ContextUnit);
    
    // Get status of a running workflow/task
    rpc GetTaskStatus(contextcore.ContextUnit) returns (contextcore.ContextUnit);
    
    // Execute a small segment of agent-generated code
    rpc ExecuteCode(contextcore.ContextUnit) returns (contextcore.ContextUnit);
}
